signer/tests/integration/transaction_coordinator.rs

```diff
@@ -123,8 +139,169 @@ async fn get_signer_public_keys_and_aggregate_key_falls_back() {
+            let ctx = TestContext::builder()
+                .with_storage(db.clone())
+                .with_mocked_clients()
+                .build();
+
+            // 2. Populate each database with the same data, so that they
+            //    have the same view of the canonical bitcoin blockchain.
+            //    This ensures that they participate in DKG.
+            data.write_to(&db).await;
+
+            (ctx, db, kp)
+        })
+        .collect::<Vec<_>>()
+        .await;
+
+    let network = network::in_memory::Network::new();
+
+    // 3. Check that there are no DKG shares in the database.
+    for (_, db, _) in signers.iter() {
+        let some_shares = db.get_last_encrypted_dkg_shares().await.unwrap();
+        assert!(some_shares.is_none());
+    }
+
+    // 4. Start the [`TxCoordinatorEventLoop`] and [`TxSignerEventLoop`]
+    //    processes for each signer.
+    let tx_coordinator_processes = signers.iter().map(|(ctx, _, kp)| TxCoordinatorEventLoop {
+        network: network.connect(),
+        context: ctx.clone(),
+        context_window: 10000,
+        private_key: kp.secret_key().into(),
+        signing_round_max_duration: Duration::from_secs(10),
+        threshold: ctx.config().signer.bootstrap_signatures_required,
+        dkg_max_duration: Duration::from_secs(10),
+    });
+
+    let tx_signer_processes = signers.iter().map(|(context, _, kp)| TxSignerEventLoop {
+        network: network.connect(),
+        threshold: context.config().signer.bootstrap_signatures_required as u32,
+        context: context.clone(),
+        context_window: 10000,
+        blocklist_checker: Some(()),
+        wsts_state_machines: HashMap::new(),
+        signer_private_key: kp.secret_key().into(),
+        rng: rand::rngs::OsRng,
+    });
+
+    // We only proceed with the test after all processes have started, and
+    // we use this counter to notify us when that happens.
+    let start_count = Arc::new(AtomicU8::new(0));
+
+    tx_coordinator_processes.for_each(|ev| {
+        let counter = start_count.clone();
+        tokio::spawn(async move {
+            counter.fetch_add(1, Ordering::Relaxed);
+            ev.run().await
+        });
+    });
+
+    tx_signer_processes.for_each(|ev| {
+        let counter = start_count.clone();
+        tokio::spawn(async move {
+            counter.fetch_add(1, Ordering::Relaxed);
+            ev.run().await
+        });
+    });
+
+    while start_count.load(Ordering::SeqCst) < 6 {
+        tokio::time::sleep(Duration::from_millis(10)).await;
+    }
+
+    // 5. Once they are all running, signal that DKG should be run. We
+    //    signal them all because we do not know which one is the
+    //    coordinator.
+    let event = SignerEvent::TxSigner(TxSignerEvent::NewRequestsHandled);
+    signers.iter().for_each(|(ctx, _, _)| {
+        ctx.get_signal_sender()
+            .send(SignerSignal::Event(event.clone()))
```

Comment:
nit: I think you could just `.send(TxSignerEvent::NewRequestsHandled.into())` here